In [19]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
%run  Data.ipynb

final

Country Code  ATMs per 1000 Adults  iPhone Users  \
0   United Arab Emirates   AE             61.909397      17943000   
1              Argentina   AR             58.841633      60664000   
2              Australia   AU            160.137780      31770000   
3                Austria   AT            119.094724      13471000   
4             Azerbaijan   AZ             35.082586      10697000   
5                Belgium   BE             93.653161      12938000   
6               Bulgaria   BG             90.889727       9195000   
7                 Brazil   BR            114.793823     257814000   
8                 Canada   CA            221.126457      29390000   
9            Switzerland   CH             97.611221      11700000   
10                 Chile   CL             56.364734      23206000   
11              Colombia   CO             40.082031      57327000   
12            Costa Rica   CR             79.614760       7536000   
13                Cyprus   CY             53.099612       1111000   
14        Czech Republic   CZ             49.561956      13925000   
15               Germany   DE            123.056697      96360000   
16               Denmark   DK             54.014077       7266000   
17                 Spain   ES            118.619050      50926000   
18               Finland   FI             35.820472       7399000   
19                France   FR            108.025692      66681000   
20                Greece   GR             59.432401      12682000   
21             Guatemala   GT             33.831861      18121000   
22               Hungary   HU             57.877075      11786000   
23             Indonesia   ID             49.482874     338426000   
24                 India   IN             17.802365    1011054000   
25               Ireland   IE             84.723645       4902000   
26                Jordan   JO             29.931674      13798000   
27                 Japan   JP            127.454390     158591000   
28            Kazakhstan   KZ             72.082918      31390000   
29              Cambodia   KH             10.877094      20851000   
30                Kuwait   KW             56.093616       8305000   
31               Lebanon   LB             37.611282       4400000   
32            Luxembourg   LU            105.258997        807000   
33               Morocco   MA             25.288598      43080000   
34                Mexico   MX             48.466025     106831000   
35              Malaysia   MY             52.174540      44111000   
36           Netherlands   NL             53.492292      20809000   
37                Norway   NO             48.305803       5841000   
38                Panama   PA             61.357130       6947000   
39                  Peru   PE             54.958746      34236000   
40           Philippines   PH             23.359515     120255000   
41                Poland   PL             63.456852      56838000   
42              Portugal   PT            176.810262      11715000   
43                 Qatar   QA             65.307808       3610000   
44               Romania   RO             65.148943      23120000   
45          Saudi Arabia   SA             70.579993      52796000   
46             Singapore   SG             59.464055       8211000   
47           El Salvador   SV             33.792877       9334000   
48                Sweden   SE             40.178787      12639000   
49              Thailand   TH            111.309382      84797000   
50                Turkey   TR             76.063025      73639000   
51          South Africa   ZA             66.251562      85197000   

    Starbucks Locations  GEI Score  
0                   144       58.8  
1                   108       22.2  
2                    22       72.5  
3                    18       63.5  
4                     4       31.1  
5                    19       63.0  
6                     5       22.7  
7                   102       20.1  
8                  1468   

In [20]:
# Convert the GEI Score into a binary output
final = final.sort_values(["GEI Score"], ascending=False)
final = final.reset_index(drop=True)

# Any country with GEI score above 42.2 is considered a potential country
final['GEI Score'] = final['GEI Score'].apply(lambda x: 1 if x >= 42.2 else 0)
final['GEI Score'] = final["GEI Score"].astype(bool)
final['GEI Score'] = final["GEI Score"].astype(int)

<img src="pics/GEI.png" width="850">

This map was exported from the GEDI website, check it out [here](http://thegedi.org/countries), and it shows that countries with an index lower than 42.2, colored in orange, were not considered top ranked countries and that is why countries above that threshold were considered awesome for expansion and were labeled 1 and all the other countries were labeled 0. 

In [21]:
final.head()

Country Code  ATMs per 1000 Adults  iPhone Users  Starbucks Locations  \
0  Switzerland   CH             97.611221      11700000                   61   
1       Canada   CA            221.126457      29390000                 1468   
2       Sweden   SE             40.178787      12639000                   18   
3      Denmark   DK             54.014077       7266000                   21   
4    Australia   AU            160.137780      31770000                   22   

   GEI Score  
0          1  
1          1  
2          1  
3          1  
4          1

In [16]:
final.tail()

Country Code  ATMs per 1000 Adults  iPhone Users  Starbucks Locations  \
47    Indonesia   ID             49.482874     338426000                  268   
48       Brazil   BR            114.793823     257814000                  102   
49  El Salvador   SV             33.792877       9334000                   11   
50    Guatemala   GT             33.831861      18121000                    7   
51     Cambodia   KH             10.877094      20851000                    4   

    GEI Score  
47          0  
48          0  
49          0  
50          0  
51          0

In [11]:
#logistic Regression Model
X = final[["ATMs per 1000 Adults", "iPhone Users", "Starbucks Locations"]]
y = final[["GEI Score"]]


X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)
logreg = LogisticRegression()
logreg.fit(X_train, y_train)

print "Logistic Regression's accuracy is", logreg.score(X_test,y_test)*100,"%"



Logistic Regression's accuracy is 44.4444444444 %


### What does that mean?

- The logistic regression model was able to classify the test data correctly 44% of the time. Although this score will have to be compared with the other models, its still considered to be a low score for a supervised learning algorithm.

- More adjustments would need to be made to the model, for example, removing outliers, feature scaling or normalization. 

